server/models/dataset_helpers/image_metadata.py

```diff
@@ -475,37 +486,55 @@ def load(cls, value, clinical, private):
 
 def _validateMetadata(clinical):
     """
-    Validate metadata with respect to consistency between fields. Fields with
-    unknown values always validate. An InconsistentValuesException is raised
-    if a value violates one of the pre-defined rules.
+    Validate metadata with respect to consistency between fields. An
+    InconsistentValuesException is raised if a value violates one of the
+    pre-defined rules. An AtypicalBenignMalignantException is raised when the
+    the value of the benign/malignant field is atypical for the value of the
+    diagnosis field.
     """
     diagnosis = clinical.get('diagnosis')
-    benignMalignant = clinical.get('benign_malignant')
-    diagnosisConfirmType = clinical.get('diagnosis_confirm_type')
+
+    if diagnosis == 'melanoma':
+        if clinical.get('benign_malignant') is None:
+            clinical['benign_malignant'] = 'malignant'
+        if clinical.get('benign_malignant') != 'malignant':
+            raise InconsistentValuesException(
+                names=[DiagnosisFieldParser.name, BenignMalignantFieldParser.name],
+                values=[diagnosis, clinical.get('benign_malignant')])
 
     if diagnosis in [
         'basal cell carcinoma',
-        'melanoma',
         'squamous cell carcinoma',
-    ] and benignMalignant is not None and benignMalignant != 'malignant':
-        raise InconsistentValuesException(
-            names=[DiagnosisFieldParser.name, BenignMalignantFieldParser.name],
-            values=[diagnosis, benignMalignant])
-
-    if diagnosis == 'nevus' and benignMalignant is not None and benignMalignant != 'benign':
-        raise InconsistentValuesException(
-            names=[DiagnosisFieldParser.name, BenignMalignantFieldParser.name],
-            values=[diagnosis, benignMalignant])
-
-    if benignMalignant in [
+    ] and clinical.get('benign_malignant') in [
+        'benign',
+        'indeterminate/benign',
+    ]:
+        raise AtypicalBenignMalignantException(
```

Comment:
We don't want to abort processing here, as the remaining rules may still with an error.